spring-cloud-kubernetes-test-support/src/main/java/org/springframework/cloud/kubernetes/integration/tests/commons/native_client/Util.java

```diff
@@ -208,20 +208,19 @@ public void deleteAndWait(String namespace, V1Deployment deployment, V1Service s
 						labelSelector(podLabels), null, null, null, null, null, null, null, null);
 				waitForDeploymentToBeDeleted(deploymentName, namespace);
 				waitForDeploymentPodsToBeDeleted(podLabels, namespace);
+			}
 
-				if (service != null) {
-					service.getMetadata().setNamespace(namespace);
-					coreV1Api.deleteNamespacedService(service.getMetadata().getName(), service.getMetadata().getNamespace(),
+			if (service != null) {
```

Comment:
it was my mistake, of course. I was doing : 

```
if (deployment != null) {

        if (service != null) { .... }
}
```

but for the external-name service, deployment is null, but service is not. 

And obviously, I was not removing it. This PR corrects this mistake.